src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -262,6 +262,10 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
             if (isset($this->aliases[$id])) {
                 $id = $this->aliases[$id];
             }
+            if (isset($this->privates[$id]) && !$this instanceof ContainerBuilder) {
```

Comment:
well, once compiled, the array is not empty anymore, but we *do* want the deprecation in this case. So please remove the extra check you added there (which was not there before moving the notice btw, so you are the one adding a difference in the behavior)